### PR TITLE
core: Get rid of SchemaRegistry (#497)

### PR DESCRIPTION
**Pull request by bscholtes1A**

### CODE REVIEW

File: extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java [hunk 1]
```diff
@@ -14,24 +14,12 @@
...
+public class S3BucketSchema {
+    
+    private S3BucketSchema() {
     }
 
-    @Override
-    public String getName() {
-        return TYPE;
-    }
+    public static final String TYPE = "AmazonS3";
```
[the remark this answers is not in the document]
Comment (bscholtes1A) [reply extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java:22]: Even if I agree it' s a bit a pity to have a whole module for that file, I think `s3-provision` should only contain stuff about S3 bucket provisioning. Here some constants are also used in the S3 data operator which is related to the data transfer and not the provisioning. 

Another approach could be to just put that file directly at the root of the `s3` module (next to the `provision` and `s3-data-operator` submodules). Don't if this is good practice tough, WDYT?

[the remark this answers is not in the document]
Comment (jimmarino) [reply extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java:22]: Aren't those things related to provisioning and it's OK to move the constants into s3-provision?

[the remark this answers is not in the document]
Comment (ndr-brt) [reply extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java:22]: Maybe we could rename `s3-schema` to `s3-core` and move there also classes from `s3-provision` that are no strictly related to provisioning like `ClientProvider` (and using it also on `s3-data-operator` that, at the moment, depends on `s3-provision`, but it seems not correct to me)

[the remark this answers is not in the document]
Comment (bscholtes1A) [reply extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java]: done!

File: extensions/azure/blobstorage/blob-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/azure/AzureBlobStoreSchema.java [hunk 2]
```diff
@@ -15,25 +15,13 @@
...
 
-import org.eclipse.dataspaceconnector.core.schema.DataSchema;
-import org.eclipse.dataspaceconnector.spi.types.domain.schema.SchemaAttribute;
+public class AzureBlobStoreSchema {
 
-public class AzureBlobStoreSchema extends DataSchema {
+    private AzureBlobStoreSchema() {
+    }
 
     public static final String TYPE = "AzureStorage";
```
[the remark this answers is not in the document]
Comment (bscholtes1A) [reply extensions/azure/blobstorage/blob-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/azure/AzureBlobStoreSchema.java:23]: done!

File: extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (bscholtes1A) [reply extensions/aws/s3/s3-schema/src/main/java/org/eclipse/dataspaceconnector/core/schema/s3/S3BucketSchema.java]: done

**Review by ndr-brt — APPROVED** [review]
Fine to me, the only thing I would do is to rename the `azure/blobstorage/provision` folder to `azure/blobstorage/blob-provision`, to be consistent with the s3 one. (maybe the same for the `api` folder).

**Comment by bscholtes1A** [comment]
> Fine to me, the only thing I would do is to rename the `azure/blobstorage/provision` folder to `azure/blobstorage/blob-provision`, to be consistent with the s3 one. (maybe the same for the `api` folder).

done

**Review by jimmarino — APPROVED** [review]

**Comment by jimmarino** [comment]
@bscholtes1A can you merge this now?

**Comment by bscholtes1A** [comment]
> @bscholtes1A can you merge this now?

yeap for me it's ready to go.

**Pull request merged**